packages/tracing-internal/src/browser/browserTracingIntegration.ts

```diff
@@ -0,0 +1,504 @@
+          op: 'pageload',
+          origin: 'auto.pageload.browser',
+          metadata: { source: 'url' },
+        };
+        browserTracingStartPageLoadSpan(context);
+      }
+
+      if (options.instrumentNavigation && client.emit) {
+        addHistoryInstrumentationHandler(({ to, from }) => {
+          /**
+           * This early return is there to account for some cases where a navigation transaction starts right after
+           * long-running pageload. We make sure that if `from` is undefined and a valid `startingURL` exists, we don't
+           * create an uneccessary navigation transaction.
+           *
+           * This was hard to duplicate, but this behavior stopped as soon as this fix was applied. This issue might also
+           * only be caused in certain development environments where the usage of a hot module reloader is causing
+           * errors.
+           */
+          if (from === undefined && startingUrl && startingUrl.indexOf(to) !== -1) {
+            startingUrl = undefined;
+            return;
+          }
+
+          if (from !== to) {
+            startingUrl = undefined;
+            if (activeSpan) {
+              DEBUG_BUILD &&
+                logger.log(`[Tracing] Finishing current transaction with op: ${spanToJSON(activeSpan).op}`);
+              // If there's an open transaction on the scope, we need to finish it before creating an new one.
+              activeSpan.end();
+            }
+            const context: StartSpanOptions = {
+              name: WINDOW.location.pathname,
+              op: 'navigation',
+              origin: 'auto.navigation.browser',
+              metadata: { source: 'url' },
+            };
+
+            browserTracingStartNavigationSpan(context);
+          }
+        });
+      }
+
+      if (markBackgroundSpan) {
+        registerBackgroundTabDetection();
+      }
+
+      if (_experiments.enableInteractions) {
+        registerInteractionListener(options, latestRouteName, latestRouteSource);
+      }
+
+      instrumentOutgoingRequests({
+        traceFetch,
+        traceXHR,
+        tracePropagationTargets,
+        shouldCreateSpanForRequest,
+        enableHTTPTimings,
+      });
+    },
+  };
+}) as IntegrationFn;
+
+export const browserTracingIntegration = defineIntegration(_browserTracingIntegration);
+
+/**
+ * Manually start a page load span.
+ * This will only do something if the BrowserTracing integration has been setup.
+ */
+export function browserTracingStartPageLoadSpan(spanOptions: StartSpanOptions): void {
+  const client = getClient();
+  if (!client || !client.emit) {
+    return;
+  }
+
+  client.emit('startPageLoadSpan', spanOptions);
+}
+
+/**
+ * Manually start a navigation span.
+ * This will only do something if the BrowserTracing integration has been setup.
+ */
+export function browserTracingStartNavigationSpan(spanOptions: StartSpanOptions): void {
```

Comment:
L (don't wanna bike shed too much about this): wdyt about `startBrowserTracingNavigationSpan` instead? Same for pageload. I think it'd resemble the `startSpan` pattern better.